### PR TITLE
Refactor JSONSerializer

### PR DESCRIPTION
**Pull request by philippjfr**
Refactors the JSONSerializer into something that makes a bit more sense to me. There are now methods to serialize/deserialize a set of parameters into a single serialized object, and another set of methods to serialize/deserialize individual parameter values.

- [x] Add tests for `serialize_parameter_value` and `deserialize_parameter_value`

### CODE REVIEW

File: param/serializer.py [hunk 1]
```diff
@@ -28,8 +28,36 @@ def schema(cls, pobj, subset=None):
 
     @classmethod
     def serialize_parameters(cls, pobj, subset=None):
+        """
+        Should serialize the parameters on a Parameterized object
```
Comment (jbednar) [inline param/serializer.py:32]: "Should"?

Comment (jbednar) [reply param/serializer.py:32]: Oh, you mean that subclasses should implement that?  I'd prefer to make that explicit:

```suggestion
        Serialize the parameters on a Parameterized object. Must be implemented by subclasses.
```

**Comment by jlstevens** [comment]
Looks fine to me. The original intent of `serialize_parameters` and `deserialize_parameters` was closer to the new `serialize_parameter_value` and `deserialize_parameter_value` methods: I agree this way is clearer though.

File: param/serializer.py (hunk 1 above)
Comment (philippjfr) [reply param/serializer.py:32]: Okay sure, will change that consistently.

**Pull request merged**